core/src/main/java/org/fao/geonet/kernel/security/openidconnect/OIDCRoleProcessor.java

```diff
@@ -329,6 +330,11 @@ public List<String> getTokenRoles(Map<String, Object> attributes, String pathToR
                 Log.debug(Geonet.SECURITY, "oidc: pathToRoles - cannot find path component named: " + path);
                 return new ArrayList<>();
             }
+            if (o instanceof String) {
+                List<String> singleGroup = new ArrayList<>();
+                singleGroup.add((String) o);
+                return singleGroup;
+            }
             if (o instanceof Map) {
                 info = (Map<String, Object>) o;
```

Comment:
this one looks strange (missing return ?)

Reply:
right, probably

Reply:
If I drop the 3 lines(338, 339, 340), the testsuite fails. I'd expect the variable info being cast from `o` is being used in the for loop, but on following iterations.

Reply:
yes - its descending into the tree.